src/lv_draw/lv_draw_rect.c

```diff
@@ -1160,18 +1160,21 @@ static void draw_outline(const lv_area_t * coords, const lv_area_t * clip, const
     /*Get the inner radius*/
     lv_area_t area_inner;
     lv_area_copy(&area_inner, coords);
-    area_inner.x1 -= dsc->outline_pad;
-    area_inner.y1 -= dsc->outline_pad;
-    area_inner.x2 += dsc->outline_pad;
-    area_inner.y2 += dsc->outline_pad;
+
+    lv_coord_t pad = dsc->outline_pad - 1;
```

Comment:
I think it should be added only if pad == 0

Reply:
This would mean that pad == 0 and pad == -1 give the same result. Is that intended?

Reply:
Yes, I think it's better to make some inconsistency on edge cases than on all cases. E.g. with your original suggestion, border_width = 2 and outline_width = 2 would result in different widths very close to each other.

Or maybe it'd be even better to no change the width at all. We can say that pad=0 means on the edge.

Reply:
I think they both have to be changed together, otherwise the switch border looks a lot smaller than it should, so I went with your original suggestion (change only when pad == 0).